Components/Payment/Provider/ActivePaymentMethodsProvider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace MollieShopware\Components\Payment\Provider;
+
+use Exception;
+use Mollie\Api\Exceptions\ApiException;
+use MollieShopware\Components\Config;
+use MollieShopware\Components\MollieApiFactory;
+use MollieShopware\Components\Payment\ActivePaymentMethodsProviderInterface;
+use Psr\Log\LoggerInterface;
+use Shopware\Models\Shop\Shop;
+
+class ActivePaymentMethodsProvider implements ActivePaymentMethodsProviderInterface
+{
+    /** @var Config */
+    private $config;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(Config $config, LoggerInterface $logger)
+    {
+        $this->config = $config;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Returns a collection of active payment methods from the Mollie API.
+     *
+     * @param array $parameters
+     * @param Shop[]|null $shops
+     *
+     * @return array
+     */
+    public function getActivePaymentMethodsFromMollie($parameters = [], $shops = [])
+    {
+        $methods = [];
+
+        if (empty($shops)) {
+            return $methods;
+        }
+
+        foreach($shops as $shop) {
+            try {
+                $methods = $this->getActivePaymentMethodsForShop($shop, $methods, $parameters);
```

Comment:
wouldnt you require a methods merge if it iterates over shops?
otherwise it would always have the ones from the last shop?